web/src/app/admin/assistants/PersonaTable.tsx

```diff
@@ -116,28 +150,30 @@ export function PersonasTable({ personas }: { personas: Persona[] }) {
               <div
                 key="is_visible"
                 onClick={async () => {
-                  const response = await fetch(
-                    `/api/admin/persona/${persona.id}/visible`,
-                    {
-                      method: "PATCH",
-                      headers: {
-                        "Content-Type": "application/json",
-                      },
-                      body: JSON.stringify({
-                        is_visible: !persona.is_visible,
-                      }),
+                  if (isEditable) {
+                    const response = await fetch(
+                      `/api/admin/persona/${persona.id}/visible`,
+                      {
+                        method: "PATCH",
+                        headers: {
+                          "Content-Type": "application/json",
+                        },
+                        body: JSON.stringify({
+                          is_visible: !persona.is_visible,
+                        }),
+                      }
+                    );
+                    if (response.ok) {
+                      router.refresh();
+                    } else {
+                      setPopup({
+                        type: "error",
+                        message: `Failed to update persona - ${await response.text()}`,
+                      });
                     }
```

Comment:
Generally prefer to place API calls outside of the JSX
Looks like this was already being done in the component but a good chance to extract it.